### PR TITLE
Allow package installation

### PR DESCRIPTION
**Pull request by larryliu0820**
Adding `pyproject.toml` and `setup.py`. Version is hardcoded and should probably change.

### CODE REVIEW

**Comment by larryliu0820** [comment]
@ev-br @honno @lezcano mind taking a look?

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+[project]
+name = "numpy_pytorch_interop"
+dependencies = [
+    "numpy",
+    "torch",
+]
+version = "0.1.0"
+[tool.setuptools.packages.find]
+where = ["."]
+include = ["autogen", "torch_np.*"]  # ["*"] by default
```
Comment (ev-br) [inline pyproject.toml:14]: autogen folder is not really needed, there's nothing worth installing really

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+exclude = ["torch_np.tests*", "torch_np.testing"]  # empty by default
```
Comment (ev-br) [inline pyproject.toml:15]: Why?

**Review by ev-br — COMMENTED** [review]
As a pure python package,  just setting PYTHONPATH is technically enough. Could you elaborate on how this makes what easier or more convenient? 

If it's needed for something, I'm on a fence tbh, so can merge and roll on, my comments can be easily addressed in a follow-up

**Comment by larryliu0820** [comment]
> As a pure python package, just setting PYTHONPATH is technically enough. Could you elaborate on how this makes what easier or more convenient?
> 
> If it's needed for something, I'm on a fence tbh, so can merge and roll on, my comments can be easily addressed in a follow-up

Yeah I thought it might be easier to install than manually setting PYTHONPATH. I don't mind setting PYTHONPATH though. Also is there a plan to publish/distribute this project to, say pypi?

File: pyproject.toml (hunk 2 above)
Comment (larryliu0820) [reply pyproject.toml]: updated.

**Comment by ev-br** [comment]
> Also is there a plan to publish/distribute this project to, say pypi?

Good question. Do you think it would be useful as a stand alone package? I'm genuinely curious on your intended use

**Comment by larryliu0820** [comment]
> > Also is there a plan to publish/distribute this project to, say pypi?
> 
> Good question. Do you think it would be useful as a stand alone package? I'm genuinely curious on your intended use

I'm working on a pytorch issue where we want to use `torch_np` in pytorch compiler. See https://github.com/pytorch/pytorch/issues/93684#issuecomment-1436026703 for more details. In the PR I'm working on, I want to check if `numpy_pytorch_interop` is installed:

"""python
try:
    import torch_np
    HAS_NUMPY_TORCH_INTEROP = True
except ModuleNotFoundError:
    torch_np = None
    HAS_NUMPY_TORCH_INTEROP = False
"""

**Comment by ev-br** [comment]
Thanks @larryliu0820 , merged.

**Comment by ev-br** [comment]
Great to hear you're working on that! Feel free to ping me if there are glitches (there surely are).

**Pull request merged**